backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -40,8 +41,13 @@ def __init__(
         self.revoked_tables = revoked_tables
         self.source_environment = source_environment
         self.target_environment = target_environment
-        self.shared_db_name = self.build_shared_db_name()
+        self.catalog_details = catalog
+        self.source_accountid = catalog.account_id if catalog else source_environment.AwsAccountId
+        self.source_region = catalog.region if catalog else source_environment.region
+        self.source_database_name = catalog.database_name if catalog else dataset.GlueDatabaseName
         self.principals = self.get_share_principals()
+        self.shared_db_name = self.build_shared_db_name()
+        self.verify_catalog_ownership()
```

Comment:
It is not considered a good practice to perform IO or raise exceptions in object contruction.. I think this should be moved to where this actually matters at runtime.

Reply:
Moved to `create_lf_processor` function after the processor object is created